nixos/modules/services/web-apps/vikunja.nix

```diff
@@ -0,0 +1,235 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.vikunja;
+  configFile = pkgs.writeText "config.json" (
+    builtins.toJSON (
+      cfg.config // cfg.extraConfig // {
+        service.frontendurl = "${cfg.frontendScheme}://${cfg.frontendHostname}";
+        }
+   ));
+  useMysql = cfg.config.database.type == "mysql";
+  usePostgresql = cfg.config.database.type == "postgres";
+in {
+  options.services.vikunja = with lib; {
+    enable = mkEnableOption "Enable Vikunja service";
+    package-api = mkOption {
+      default = pkgs.vikunja-api;
+      type = types.package;
+      defaultText = "pkgs.vikunja-api";
+      description = "vikunja-api derivation to use.";
+    };
+    package-frontend = mkOption {
+      default = pkgs.vikunja-frontend;
+      type = types.package;
+      defaultText = "pkgs.vikunja-frontend";
+      description = "vikunja-frontend derivation to use.";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "vikunja";
+      description = "User account under which vikunja runs. The user will only be created when using the default.";
+    };
+    stateDir = mkOption {
+      type = types.str;
+      default = "/var/lib/vikunja";
+      description = "Vikunja data directory";
+    };
+    environmentFiles = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = "List of environment files set in the vikunja systemd service/";
+    };
+    setupNginx = mkOption {
+      type = types.bool;
+      default = config.services.nginx.enable;
+      description = "Whether to setup NGINX.";
+    };
+    frontendScheme = mkOption {
+      type = types.enum [ "http" "https" ];
+      description = "Whether the site is available via http or https. This does not configure https or ACME in nginx!";
+    };
+    frontendHostname = mkOption {
+      type = types.str;
+      description = "The Hostname under which the frontend is running.";
+    };
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = {};
+      description = "Extra config that fill directly be passed into the Vikunja config.";
+    };
+    config = {
```

Comment:
Thanks for the idea with the configuration options. I took a bit of time to build a NixOS test, but since it's my first NixOS test, I'm really not sure if I did it well, so feel free to suggest improvements,

Reply:
Should I add `passthru.tests = nixosTests.vikunja` to the vikunja-api or vikunja-frontend package or both? The NixOS test mostly tests the api but also a bit of the frontend.

Reply:
Hey @aanderse could you have a look at this question?

Reply:
I'm sorry @em0lar I missed this comment. Thanks for pinging me. Sounds like it makes sense to add to both.

Reply:
Thank you @aanderse for your reply. I added it to both packages